src/sagemaker/jumpstart/curated_hub/sync/comparator.py

```diff
@@ -0,0 +1,78 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module provides comparators for syncing s3 files."""
+from __future__ import absolute_import
+from datetime import timedelta
+from sagemaker.jumpstart.constants import JUMPSTART_LOGGER
+
+from sagemaker.jumpstart.curated_hub.types import FileInfo
+
+
+class BaseComparator:
+    """BaseComparator object to be extended."""
+
+    def determine_should_sync(self, src_file: FileInfo, dest_file: FileInfo) -> bool:
+        """Custom comparator to determine if src file and dest file are in sync."""
+        raise NotImplementedError
+
+
+class SizeAndLastUpdatedComparator(BaseComparator):
+    """Comparator that uses file size and last modified time.
+
+    Uses file size (bytes) and last_modified_time (timestamp) to determine sync.
+    """
+
+    def determine_should_sync(self, src_file: FileInfo, dest_file: FileInfo) -> bool:
+        """Determines if src file should be moved to dest folder."""
+        same_size = self.compare_size(src_file, dest_file)
+        is_newer_dest_file = self.compare_file_updates(src_file, dest_file)
+        should_sync = (not same_size) or (not is_newer_dest_file)
+        if should_sync:
+            JUMPSTART_LOGGER.warning(
+                "syncing: %s -> %s, size: %s -> %s, modified time: %s -> %s",
```

Comment:
why is this a warning?

Reply:
I'll address all logging in the next PR. This was set to warning for testing, but should be info